### PR TITLE
Report errors even with strict mode

### PR DESCRIPTION
**Pull request by ShadowDancer**
## What does the pull request do?
This PR allow compiler to report errors in non-strict mode
Also removes some unnecessary parameters (old members are obsoleted).

## What is the current behavior?
Currently there are ifs in code, which check if there is strict mode and throw exceptions, or do nothing. In such situation it is impossible to gather diagnostics, because all errors are ignored.

## What is the updated/expected behavior with this PR?
StrictMode is unchanged. Non strict mode is unchanged, but not thrown exceptions are stored to be analyzed for later.

## How was the solution implemented (if it's not obvious)?
`AstTransformationContext` is augmented with list of error and new `Error` method, which throws or adds error to list

## Breaking changes
None, made `StrictMode` obsolete, so transformers use `Error` method, instead of handling strict mode manually. Also obsoleted members with strict mode passed as parameter (it was passed anyway in context, no reason to pass it twice).

### CODE REVIEW

**Comment by ShadowDancer** [comment]
@kekekeks this is ready for review.

File: src/XamlX/Transform/Transformers/XArgumentsTransformer.cs [hunk 1]
```diff
@@ -15,9 +15,10 @@ public IXamlAstNode Transform(AstTransformationContext context, IXamlAstNode nod
                 var argDirectives = ni.Children.OfType<XamlAstObjectNode>()
                     .Where(d => d.Type is XamlAstXmlTypeReference xref
                                 && xref.XmlNamespace == XamlNamespaces.Xaml2006 && xref.Name == "Arguments").ToList();
-                if (argDirectives.Count > 1 && context.StrictMode)
```
Comment (kekekeks) [inline src/XamlX/Transform/Transformers/XArgumentsTransformer.cs:18]: The exception has served as an "early escape" here. We need to be sure that node processing is properly halted after error is reported.

**Review by kekekeks — CHANGES_REQUESTED** [review]
I think `context.Error` call should be strictly followed by `return` statement. There are places in the code that are technically checking for the strict mode, but running in non-strict mode was **not tested at all**

File: src/XamlX/Transform/Transformers/XArgumentsTransformer.cs (hunk 1 above)
Comment (ShadowDancer) [reply src/XamlX/Transform/Transformers/XArgumentsTransformer.cs:18]: Fixed

**Comment by ShadowDancer** [comment]
@kekekeks updated this PR as you demanded.

I thought that execution could keep going, i.e. validate attached properties even if control type is unknown, but I guess it would make too much mess and little gain after all.

**Pull request closed without merging**